packages/core/src/storages/key_value_store.ts

```diff
@@ -141,16 +144,30 @@ export class KeyValueStore {
      * @param key
      *   Unique key of the record. It can be at most 256 characters long and only consist
      *   of the following characters: `a`-`z`, `A`-`Z`, `0`-`9` and `!-_.'()`
+     * @param defaultValue
+     *   Fallback that will be returned if no value if present in the storage.
      * @returns
      *   Returns a promise that resolves to an object, string
      *   or [`Buffer`](https://nodejs.org/api/buffer.html), depending
      *   on the MIME content type of the record.
      */
-    async getValue<T = unknown>(key: string): Promise<T | null> {
+    async getValue<T = unknown>(key: string, defaultValue?: T): Promise<T | null> {
         ow(key, ow.string.nonEmpty);
         const record = await this.client.getRecord(key);
 
-        return record?.value as T ?? null;
+        return record?.value as T ?? defaultValue ?? null;
+    }
+
+    async getAutoSavedValue<T extends Dictionary = Dictionary>(key: string, defaultValue = {} as T): Promise<T> {
+        if (this.cache.has(key)) {
+            return this.cache.get(key) as T;
+        }
+
+        const value = await this.getValue<T>(key, defaultValue);
+        this.cache.set(key, value);
+        this.config.getEventManager().on('persistState', () => this.setValue(key, value));
```

Comment:
Do we want to add a listener every time someone sets a value? Can we store references instead and call the listener once?

Reply:
wdym? we only add it once, the value is cached, so this branch will be executed once per key only (for a given instance of KV store)

the use case here is to have persistent value that will be always saved on the persist state event, which is (among other cases) fired during actor migration on platform

Reply:
> especially if someone calls that method with the same key multiple times

That's why we cache the value, as I said, this branch will be executed only once and then the cached reference will be used instead for following calls.

Usually there will be just one value here (so one listener), the crawler state.

Reply:
Woops, yeah sorry I was expressing myself wrong (meant multiple different keys), but it should be fine nevertheless. Just a thought